File: cpp/remove_duplicates_from_sorted_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        ListNode *tmp = head;
        while (tmp)
        {
            if (!tmp->next)
                break;
            else
                if (tmp->val == tmp->next->val)
                {
                    tmp->next = tmp->next->next;
                }
                else
                    tmp = tmp->next;
        }
        return head;
    }
};

File: python/remove_duplicates_from_sorted_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param head, a ListNode
    # @return a ListNode
    def deleteDuplicates(self, head):
        tmp = head
        while tmp != None:
            if tmp.next == None:
                break
            else:
                if tmp.val == tmp.next.val:
                    tmp.next = tmp.next.next
                else:
                    tmp = tmp.next
        return head